src/sparseml/pytorch/sparsification/quantization/quantize.py

```diff
@@ -0,0 +1,140 @@
+            "Observers constructor excluding quantization range or symmetry"
+        ),
+    )
+
+    @classmethod
+    def default_activation_args(cls):
+        """
+        :return: default 8 bits asymmetric settings
+        """
+        return cls(num_bits=8, symmetric=False)
+
+    @classmethod
+    def default_weight_args(cls):
+        """
+        :return: default 8 bits symmetric settings
+        """
+        return cls(num_bits=8, symmetric=True)
+
+    def get_observer(self) -> "torch.quantization.FakeQuantize":
+        """
+        :return: torch quantization FakeQuantize built based on these QuantizationArgs
+        """
+        qscheme = (
+            torch.per_tensor_symmetric if self.symmetric else torch.per_tensor_affine
+        )
+        target_dtype = torch.qint8
+        quant_min, quant_max = compute_range(target_dtype, self.num_bits)
+        return torch_quantization.FakeQuantize.with_args(
+            observer=torch_quantization.MovingAverageMinMaxObserver,
+            quant_min=quant_min,
+            quant_max=quant_max,
+            dtype=target_dtype,
+            qscheme=qscheme,
+            **self.kwargs,
+        )
+
+
+class QuantizationScheme(BaseModel):
+    """
+    Class composed of QuantizationArgs to build QConfig and QuantWrapper objects for
+    quantizing models. Provides a simple user interface for defining how inputs,
+    weights, and outputs should be quantized
+    """
+
+    input_activations: Optional[QuantizationArgs] = Field(
+        default_factory=QuantizationArgs.default_activation_args,
+        help=(
+            "target quantization setting for input activations. Set to None to "
+            "not quantize input activations. Default is 8 bits asymmetric"
+        ),
+    )
+    weights: Optional[QuantizationArgs] = Field(
+        default_factory=QuantizationArgs.default_weight_args,
+        help=(
+            "target quantization setting for model weights. Set to None to "
+            "not quantize weights. Default is 8 bits symmetric"
+        ),
+    )
+    output_activations: Optional[QuantizationArgs] = Field(
+        default=None,
+        help=(
+            "target quantization setting for output activations. Set to None to "
+            "not quantize output activations. Default is None"
+        ),
+    )
+
+    def get_qconfig(self) -> "torch.quantization.QConfig":
+        """
+        :return: QConfig for Modules (output activations used,
+            use QuantWrapper for inputs)
+        """
+        return _get_qconfig(self.output_activations, self.weights)
+
+    def get_wrapper_qconfig(self) -> "torch.quantization.QConfig":
+        """
+        :return: QConfig for QuantWrapper objets (input activations used)
+        """
+        return _get_qconfig(self.input_activations, self.weights)
+
+
+def _get_qconfig(
+    activation_args: Optional[QuantizationArgs], weight_args: Optional[QuantizationArgs]
```

Comment:
If the argument is optional, its default value should be specified as `None`.

```suggestion
    activation_args: Optional[QuantizationArgs] = None, weight_args: Optional[QuantizationArgs] = None
```